engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -3516,6 +3517,16 @@ private void checkPlatformCompatible(AppPackage ap)
     }
   }
 
+  private void checkJavaVersionCompatible(AppPackage ap)
+  {
+    String buildJdkVersion = ap.getBuildJdkVersion();
+    String systemVersion = System.getProperty("java.version");
+    if (VersionInfo.compare(buildJdkVersion, systemVersion) > 0) {
```

Comment:
Currently you are calling this from the LaunchCommand.execute() so throwing an exception makes sense. But we can (and should) also enhance the GetAppPackageInfoCommand to display the Build-jdk value for user's benefit. In that case you may want to refactor this function to just return the 2 values (when called from GetAppPackageInfoCommand) or throw an exception (like here when called from  LunchCommand).